shared/chat/inbox/container/remote.js

```diff
@@ -17,14 +57,44 @@ export type RemoteConvMeta = $Diff<
   {onSelectConversation: () => void}
 >
 
-// To cache the list
-const valuesCached = memoize((...vals) => vals.map(v => v))
-
-const metaMapToFirstValues = memoize(metaMap =>
-  metaMap
-    .partialSort(maxShownConversations, (a, b) => b.timestamp - a.timestamp)
-    .filter((_, id) => Constants.isValidConversationIDKey(id))
-    .valueSeq()
+// Sort by timestamp
+const getSortedConvMetas = Container.createSelector([getMetas], ([map, state]) =>
+  map
+    .sort((a, b) => b.timestamp - a.timestamp)
+    .slice(0, maxShownConversations)
+    .toList()
+    .map((m): RemoteConvMeta => {
+      const hasUnread = Constants.getHasUnread(state, m.conversationIDKey)
+      const styles = Constants.getRowStyles(m, false, hasUnread)
+      const participantNeedToRekey = m.rekeyers.size > 0
+      const _username = state.config.username || ''
+      const youNeedToRekey = !!participantNeedToRekey && m.rekeyers.has(_username)
+      return {
+        backgroundColor: Styles.globalColors.white,
+        channelname: m.channelname,
+        conversationIDKey: m.conversationIDKey,
+        hasBadge: Constants.getHasBadge(state, m.conversationIDKey),
+        hasResetUsers: !!m.resetParticipants && m.resetParticipants.size > 0,
+        hasUnread,
+        iconHoverColor: styles.iconHoverColor,
+        isFinalized: !!m.wasFinalizedBy,
+        isInWidget: true,
+        isMuted: m.isMuted,
+        isSelected: false,
+        // excluding onSelectConversation
+        participantNeedToRekey,
+        participants: m.teamname ? [] : Constants.getRowParticipants(m, _username).toArray(),
+        showBold: styles.showBold,
+        snippet: m.snippet,
+        snippetDecoration: m.snippetDecoration,
+        subColor: styles.subColor,
+        teamname: m.teamname,
+        timestamp: Constants.timestampToString(m),
+        usernameColor: styles.usernameColor,
+        youAreReset: m.membershipType === 'youAreReset',
+        youNeedToRekey,
+      }
+    })
```

Comment:
Let me have a look

Reply:
so i think we should avoid these types of deeply nested functions. you can collapse this with just returning early, something like
```
    .filter((meta, id) => {
      if (!Constants.isValidConversationIDKey(id)) {
        return false
      }
      if (meta.teamType === 'adhoc') {
        return !meta.isMuted
      } 
      if (meta.teamType !== 'big') {
        return false
      }
      // channels
      if (meta.notificationsGlobalIgnoreMentions) {
        return false
      }
      // @here and @mention not ignored
      if (meta.notificationsDesktop === 'onAnyActivity') {
        return true
      } 
      if (meta.notificationsDesktop === 'onWhenAtMentioned') {
          // check if user has been mentioned in the message
          let snippet = meta.snippet.toLowerCase()
          return (
            (!!_lastState.config.username && snippet.indexOf(`@${_lastState.config.username}`) === 0) ||
            snippet.indexOf('you') === 0
          )
      }
      return false
    })
```
(check that logic cause it could be wrong)